### PR TITLE
Feature/KITOSUDV-3422

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

**Comment by a-naskret** [comment]
Depends on https://github.com/Strongminds/kitos/pull/812, after 3421 is merged this one is ready for review

**Comment by a-naskret** [comment]
@mrjsawdk ready

File: Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		LockedOutDate = GETDATE(),
+		Email = CONVERT(NVARCHAR(36), NEWID()) + '_deleted_user@kitos.dk',
+		PhoneNumber = null,
+		LastName = (CASE
+					WHEN LastName IS NULL THEN ''
+					ELSE LastName
+				   END) + ' (SLETTET)',
+		DeletedDate = GETDATE(),
+		Deleted = 1,
+		IsGlobalAdmin = 0,
```
[the remark this answers is not in the document]
Comment (mrjsawdk) [reply Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql:22]: Procudure..
- select a user which you expect and not expect to be deleted.
- run the migration
- validate that only the expected side effects happen

File: Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Content:
+    Sets Users with no assossiated Organizations as deleted
+*/
+
+BEGIN
+	UPDATE [User]
+	SET 
+		Name = 'Slettet bruger',
+		LockedOutDate = GETDATE(),
+		Email = CONVERT(NVARCHAR(36), NEWID()) + '_deleted_user@kitos.dk',
```
Comment (mrjsawdk) [inline Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql:14]: I think we should also do what the "delete command" does (in code) which covers:
- Remove all sso identities
- Remove all system roles
- Remove all contract roles
- Remove all dpr roles
- Remove all org unit roles

**Review by mrjsawdk — CHANGES_REQUESTED** [review]

**Comment by a-naskret** [comment]
@mrjsawdk ready for review

File: Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	(
+		Id int
+	)
+
+	INSERT INTO #get_userIds_to_delete
+	SELECT T0.Id
+	FROM [User] T0 
+	LEFT JOIN  OrganizationRights T1 
+	ON T0.Id = T1.UserId
+	WHERE T1.Id IS NULL AND T0.Deleted = 0
```
Comment (mrjsawdk) [inline Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql]: do not delete global admins

**Review by mrjsawdk — APPROVED** [review]

File: Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql (hunk 3 above)
Comment (mrjsawdk) [reply Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_Users_Not_Associated_With_Any_Org.sql]: i fixed that

**Review by mrjsawdk — APPROVED** [review]

**Pull request merged**